tests/shared/src/test/scala/cats/effect/std/HotswapSpec.scala

```diff
@@ -104,6 +107,46 @@ class HotswapSpec extends BaseSpec { outer =>
 
       go must completeAs(())
     }
+
+    "not block current resource while swap is instantiating new one" in ticked {
+      implicit ticker =>
+        val go = Hotswap.create[IO, Unit].use { hs =>
+          hs.swap(IO.sleep(1.minute).toResource).start *>
+            IO.sleep(5.seconds) *>
+            hs.get.use_.timeout(1.second) *> IO.unit
+        }
+        go must completeAs(())
+    }
+
+    "successfully cancel during swap and run finalizer if cancelation is requested while waiting for get to release" in ticked {
+      implicit ticker =>
+        val go = Ref.of[IO, List[String]](List()).flatMap { log =>
+          Hotswap[IO, Unit](logged(log, "a")).use {
+            case (hs, _) =>
+              for {
+                _ <- hs.get.evalMap(_ => IO.sleep(1.minute)).use_.start
+                _ <- IO.sleep(2.seconds)
+                _ <- hs.swap(logged(log, "b")).timeoutTo(1.second, IO.unit)
+                value <- log.get
+              } yield value
+          }
+        }
+
+        go must completeAs(List("open a", "open b", "close b"))
+    }
+
+    "swap is safe to concurrent cancelation" in ticked { implicit ticker =>
+      val go = IO.ref(false).flatMap { open =>
+        Hotswap[IO, Unit](Resource.unit)
+          .use {
+            case (hs, _) =>
+              hs.swap(Resource.make(open.set(true))(_ => open.set(false)))
```

Comment:
Even better, let's make sure we are only closing it if it is open ...

```suggestion
              hs.swap(Resource.make(open.set(true))(_ => open.getAndSet(false).map(_ should beTrue).void))
```